doc/development-guide.md

```diff
@@ -0,0 +1,53 @@
+# Development Guide
```

Comment:
I think a guide needs to start from the beginning (setting up dev environment) etc.

Go version needed, issue/PR creation, support, where's the development requirements link?

Reply:
Well, couldn't agree more. Can save us plenty of time!